payjoin/src/core/error.rs

```diff
@@ -75,6 +76,8 @@ pub(crate) enum InternalReplayError<SessionState, SessionEvent> {
     NoEvents,
     /// Invalid initial event
     InvalidEvent(Box<SessionEvent>, Option<Box<SessionState>>),
+    /// Session is expired
+    Expired(crate::time::Time),
```

Comment:
Whoops, I misread this to be in the SessionEvent. I think this is fine as-is. It's nice to print when it expired if an error is thrown. Would be odd to fetch that information from inside the state machine instead.